lock-api/src/main/java/com/palantir/lock/client/LeasingTimelockClient.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.palantir.lock.v2.IdentifiedTime;
+import com.palantir.lock.v2.IdentifiedTimeLockRequest;
+import com.palantir.lock.v2.ImmutableLockImmutableTimestampResponse;
+import com.palantir.lock.v2.ImmutableLockResponse;
+import com.palantir.lock.v2.ImmutableStartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.LeasableLockResponse;
+import com.palantir.lock.v2.LeasableRefreshLockResponse;
+import com.palantir.lock.v2.LeasableStartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.StartAtlasDbTransactionResponse;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.WaitForLocksRequest;
+import com.palantir.lock.v2.WaitForLocksResponse;
+import com.palantir.timestamp.TimestampRange;
+
+public final class LeasingTimelockClient implements TimelockService {
+    private final TimelockRpcClient delegate;
+
+    private LeasingTimelockClient(TimelockRpcClient timelockRpcClient) {
+        this.delegate = timelockRpcClient;
+    }
+
+    public static LeasingTimelockClient create(TimelockRpcClient timelockRpcClient) {
+        return new LeasingTimelockClient(timelockRpcClient);
+    }
+
+    @Override
+    public long getFreshTimestamp() {
+        return delegate.getFreshTimestamp();
+    }
+
+    @Override
+    public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
+        return delegate.getFreshTimestamps(numTimestampsRequested);
+    }
+
+    @Override
+    public LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request) {
+        return delegate.lockImmutableTimestamp(request);
+    }
+
+    @Override
+    public StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLockRequest request) {
+        return delegate.startAtlasDbTransaction(request);
+    }
```

Comment:
We need to either remove lockImmutableTimestamp and startAtlasDbTransaction or return leasedLocks to prevent class cast exceptions

Reply:
can't we write lockImmutableTimestamp in terms of startIdentifiedAtlasDbTransaction?

Reply:
removing the legacy startAtlasDbTransaction method seems sensible; we don't ever use it and it's clearly scoped to AtlasDB; but locking the immutable timetamp is something people sometimes do